README.md

```diff
@@ -80,24 +83,184 @@ After you input your application credentials, you can click the setup button, wh
 
 **Important**: Make sure your Resources/MobiledgeXSettings.asset file has the correct information for your application. 
 
-Once that setup has been completed, you can very easily call all the necessary API requests to connect to a cloudlet with your application deployed. Here is some example code using the MobiledgeXIntegration class that comes with the package : 
+
+### Example Usage:
+Once that setup has been completed, you can very easily call all the necessary API requests to connect to a cloudlet with your application deployed. Here is some example code using the MobiledgeXIntegration class that comes with the package 
+
+
+**Getting Edge Connection Url**
+
+MobiledgeX SDK uses the device Location and [the device MCC-MNC code](https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine) to connect you to the closest Edge cloudlet were you application instance is deployed.
+
+If your carrier is not supported yet by MobiledgeX it will throw a DmeDnsException and it will connect you to  [wifi.dme](https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine) which will connect you to the closest [regional DME](https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine).
 
 ```csharp
 using MobiledgeX;
+using DistributedMatchEngine;
+using UnityEngine;
+using System.Collections;
+
+[RequireComponent(typeof(MobiledgeX.LocationService))]
+public class YourClassName : MonoBehaviour
+{ 
+    IEnumerator Start()
+    {
+        yield return StartCoroutine(MobiledgeX.LocationService.EnsureLocation()); // Location is needed to connect you to the closet edge
+        GetEdgeConnection();
+    }
+     async void GetEdgeConnection()
+    {
+        MobiledgeXIntegration mxi = new MobiledgeXIntegration();
+        try
+        {
+            await mxi.RegisterAndFindCloudlet();
+        }
+        catch(DmeDnsException)
+        {
+            mxi.UseWifiOnly(true); // if you carrier is not supported yet, WifiOnly will connect you to wifi.dme 
+            await mxi.RegisterAndFindCloudlet();
+        }
+
+        mxi.GetAppPort(LProto.L_PROTO_HTTP); // Get the port of the desired protocol
+        string url = mxi.GetUrl("http"); // Get the url of the desired protocol
+    }
+    
+}
+```
+
+If your device doesn't have MCC-MNC code (no sim card - for ex. Oculus device), Please use UseWifiOnly before RegisterAndFindCloudlet.
```

Comment:
minor: have *a* MCC-MNC. Also instead of code I would use ID since MCC stands for Mobile Country Code. so as is, it would read as mobile country code code.